at_secondary/at_persistence_secondary_server/lib/src/keystore/hive_keystore.dart

```diff
@@ -288,19 +308,23 @@ class HiveKeystore implements SecondaryKeyStore<String, AtData?, AtMetaData?> {
 
   @override
   Future<AtMetaData?> getMeta(String key) async {
-    try {
-      var hive_key = keyStoreHelper.prepareKey(key);
-      var value = await persistenceManager.getBox().get(hive_key);
-      if (value != null) {
-        return value.metaData;
-      }
-    } on Exception catch (exception) {
-      logger.severe('HiveKeystore getMeta exception: $exception');
-      throw DataStoreException('exception in getMeta: ${exception.toString()}');
-    } on HiveError catch (error) {
-      await _restartHiveBox(error);
-      logger.severe('HiveKeystore getMeta error: $error');
-      throw DataStoreException(error.message);
+    // try {
+    //   var hive_key = keyStoreHelper.prepareKey(key);
+    //   var value = await persistenceManager.getBox().get(hive_key);
+    //   if (value != null) {
+    //     return value.metaData;
+    //   }
+    // } on Exception catch (exception) {
+    //   logger.severe('HiveKeystore getMeta exception: $exception');
+    //   throw DataStoreException('exception in getMeta: ${exception.toString()}');
+    // } on HiveError catch (error) {
+    //   await _restartHiveBox(error);
+    //   logger.severe('HiveKeystore getMeta error: $error');
+    //   throw DataStoreException(error.message);
+    // }
+    // return null;
+    if (_metaDataCache.containsKey(key)) {
+      return _metaDataCache[key];
```

Comment:
Do we need a fallback to get from hive? Donno. Let's discuss this with @gkc.

Reply:
Not yet. Commit log key-to-commit-id map is entirely in memory; let's do the same here for now. It will be simple later to switch to an LRU cache which has a configurable maximum size and reads from hive on cache hits